source/plugin/permissions.rst

```diff
@@ -2,9 +2,296 @@
+    Owner: MyPlugin v1.2.3
+
+Template-result
+~~~~~~~~~~~~~~~
+
+.. code-block:: text
+    
+    myPlugin.commands.teleport.worlds.<World>
+    
+    Description: Allows the user to teleport to the world <World>.
+    Role: staff
+    Owner: MyPlugin v1.2.3
+
+.. tip::
+
+    You might skip writing descriptions for some parent permission groups such as ``myPlugin.commands.teleport.worlds``
+    or ``myPlugin.commands`` as their meaning can be derived from the permission structure and the defined children
+    alone.
+
+Subject
+=======
+
+A ``Subject`` is a holder of assigned permissions. It can be obtained from the ``PermissionService`` via 
+``SubjectCollections``.
+``CommandSources`` such as ``Players`` are ``Subjects`` by default, but there are many other types of ``Subjects``.
+Anything that has permissions is a Subject even if it just delegates the checks to a contained Subject.
+Permissions can be granted or denied to a Subject. If a permission is neither granted nor denied its setting will be
+inherited. See Inheritance.
+Subjects provide methods to check whether they have a certain permission or not.
+Plugins that use this method should only query for the specific permission they want to check. It is the
+PermissionService's task to respect the permission and subject inheritance.
+
+Example
+~~~~~~~
+
+The following example could be used to check whether the Player is allowed to execute the teleport command.
+
+.. code-block:: java
+    
+    public canTeleport(Player subject, World targetWorld) {
+        return subject.hasPermission("myPlugin.command.teleport.execute") 
+                && (subject.getWorld() == targetWorld
+                        || subject.hasPermission("myPlugin.command.teleport." + targetWorld.getName()));
+    }
+
+Inheritance
+~~~~~~~~~~~
+
+If a ``Subject`` has a permission assigned, it will use that value.
+Otherwise it will be inherited from any parent ``Subject``. It does not matter what kind of parent (ex group or player)
+``Subject`` that might be.
+
+If neither the subject itself nor any parent subjects grant or deny a permission then it will be inherited from the
+default ``Subjects``. Each ``SubjectCollection`` defines its own defaults. The global and weakest default subject can be
+obtained from the ``PermissionService``. It is recommended (but not mandatory) that plugins define their own permissions
+to the default's transient ``SubjectData`` during every server start-up. This allows the server owner to overwrite the
+defaults defined by plugins according to his needs using the default's persistent ``SubjectData``.
+
+.. note::
+
+    Its only specified for the default ``Subjects`` that the persistent ``SubjectData`` takes precedence over transient
+    ones.
+
+If neither the Subject, nor any of its parents, nor the defaults grant and deny a permission, then it is automatically
+denied.
+
+.. note::
+
+    Order of precedence in descending order:
+    
+    * Subject itself
+    * Parent Subjects
```

Comment:
That's exactly what has been said already, just in slightly different words. A server owner can override the transient defaults, by adding a default that persists. :wink: